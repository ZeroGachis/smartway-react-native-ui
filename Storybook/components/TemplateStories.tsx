import type { Meta, StoryObj } from '@storybook/react-native';
import React from 'react';
import { StyleSheet, View } from 'react-native';

// fake component must be replace by one from the library
const Component = (props: React.PropsWithChildren) => {
    const { children } = props;
    return <>{children}</>;
}; // must be removed

type ComponentProps = React.ComponentProps<typeof Component>;

export default {
    title: 'components/Component', // change it with your component name eg('components/AwesomeButton',)
    component: Component,
    //default value
    args: {},
    // add some control in argTypes
    // some example
    // https://github.com/storybookjs/react-native/blob/next/examples/expo-example/components/ControlExamples/ControlExample/ControlExample.stories.tsx
    argTypes: {},
    // use it to wrap your storie in composant
    decorators: [
        (Story) => {
            const styles = StyleSheet.create({
                container: { paddingTop: 16 },
            });
            return (
                <View style={styles.container}>
                    <Story />
                </View>
            );
        },
    ],
} as Meta<ComponentProps>;

type Story = StoryObj<ComponentProps>;

// first story use it with args to change load only props value
// you can also use render as
// render : (args) => <AwesomeButton {...args}>
export const Default: Story = {
    args: {},
};
Default.parameters = { noSafeArea: false };
