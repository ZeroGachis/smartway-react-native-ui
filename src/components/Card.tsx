import type { ReactNode } from "react";
import React from "react";
import { StyleProp, View, ViewStyle } from "react-native";
import { useTheme } from "smartway-react-native-ui";

interface Props {
    children?: ReactNode,
    style: ViewStyle
};

export const Card = ({children, style}: Props) => {
    const theme = useTheme();
    const cardStyle: StyleProp<ViewStyle> = {
        backgroundColor: theme.sw.colors.neutral[50],
        width: '100%',
        alignItems: 'center',
        borderRadius: 12,
        paddingTop: theme.sw.spacing.xl,
        paddingBottom: theme.sw.spacing.xl,
        paddingLeft: theme.sw.spacing.m,
        paddingRight: theme.sw.spacing.m,
        ...style
        // TODO: Implement shadows
    };

    return (
        <View style={cardStyle}>
            {children}
        </View>
    );
};
