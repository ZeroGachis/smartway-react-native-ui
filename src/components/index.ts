import { Button } from 'react-native-paper';
import Logo from './logos/logos';

export { Button, Logo };