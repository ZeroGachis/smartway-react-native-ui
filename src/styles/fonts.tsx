export const fontConfig = {
    fontFamily: 'Roboto'
};