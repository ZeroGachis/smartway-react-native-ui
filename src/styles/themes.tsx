import React, { ReactNode } from 'react';
import {
    configureFonts,
    MD3LightTheme,
    Provider as PaperProvider,
    useTheme as usePaperTheme,
} from 'react-native-paper';
import { fontConfig } from './fonts';

const theme = {
    ...MD3LightTheme,
    fonts: configureFonts({ config: fontConfig }),
    sw: {
        colors: {
            primary: {
                800: '#225348',
                600: '#017F63',
                400: '#18A586',
                200: '#31E3BC',
                50: '#C5FCF0',
            },
            neutral: {
                900: '#1A2026',
                800: '#212B36',
                700: '#2F4158',
                600: '#637381',
                500: '#919EAB',
                400: '#E5EAEE',
                300: '#ECF0F4',
                200: '#F4F6F8',
                100: '#F9FAFB',
                50: '#FFFFFF',
            },
            error: {
                main: '#FF5630',
                50: '#FFE9D5',
            },
        },
        spacing: {
            xs: 6,
            s: 12,
            m: 16,
            l: 24,
            xl: 32,
            xxl: 64,
        },
    },
};

export type ThemType = typeof theme;
export const useTheme = () => usePaperTheme<ThemType>();

type Props = {
    children?: ReactNode;
};

export const ThemeProvider = ({ children }: Props) => {
    return <PaperProvider theme={theme}>{children}</PaperProvider>;
};
