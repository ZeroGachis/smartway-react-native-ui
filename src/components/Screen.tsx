import React from 'react';
import { SafeAreaView, ViewStyle, StatusBar, StyleSheet } from 'react-native';
import { useSafeAreaInsets } from 'react-native-safe-area-context';
import { useTheme } from '../styles/themes';
import DeviceInfo from 'react-native-device-info';
import type { WithTestID } from 'src/shared/type';

type Props = WithTestID<{
    children?: React.ReactNode;
    style?: ViewStyle;
    statusBarColor?: string;
}>;

export const Screen = ({ children, style, testID, statusBarColor }: Props) => {
    const theme = useTheme();
    const insets = useSafeAreaInsets();
    const isTablet = DeviceInfo.isTablet();

    const styles = StyleSheet.create({
        screen: {
            flex: 1,
            backgroundColor: theme.sw.color.neutral[0],
            marginTop: insets.top,
            paddingLeft: isTablet ? theme.sw.spacing.s : 0,
            paddingRight: isTablet ? theme.sw.spacing.s : 0,
            ...style,
        },
    });

    return (
        <SafeAreaView style={styles.screen} testID={testID}>
            <StatusBar
                translucent
                backgroundColor={
                    statusBarColor
                        ? statusBarColor
                        : theme.sw.color.neutral[0]
                }
                barStyle='dark-content'
            />
            {children}
        </SafeAreaView>
    );
};
