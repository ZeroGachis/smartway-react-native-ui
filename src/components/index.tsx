import { Button } from './buttons/Button';
import { IconButton } from './buttons/IconButton';
import { Logo } from './logos/Logo';
import { Screen } from './Screen';
import { Headline } from './typography/Headline';
import { Body } from './typography/Body';
import { Modal } from './modals/Modal';
import { TextField } from './fields/TextField';
import { Icon } from './icons/Icon';
import { TopAppBar } from './topAppBar/TopAppBar';
import { Card } from './card/Card';
import { Tag } from './tag/Tag';
import { Divider } from './divider/Divider';
import { LineChart } from './charts/LineChart';
import { Badge } from './badge/Badge';
import { DateField } from './dateSelector/DateField';
import { DateSelector } from './dateSelector/DateSelector';
import { Label } from './typography/Label';
import { FieldState, NumberField } from './numberField/NumberField';

export { default as Snackbar, useSnackbar } from './alert/Snackbar';
export { default as Banner, useBanner } from './alert/Banner';

export {
    Body,
    Button,
    IconButton,
    Modal,
    Headline,
    Icon,
    Logo,
    Label,
    Screen,
    TextField,
    TopAppBar,
    NumberField,
    FieldState,
    Card,
    Tag,
    Divider,
    LineChart,
    Badge,
    DateField,
    DateSelector,
};
