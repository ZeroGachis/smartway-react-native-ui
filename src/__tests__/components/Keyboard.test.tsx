import React from 'react';
import { act, fireEvent, render } from '@testing-library/react-native';
import { ThemeProvider } from '../../styles/themes';
import { Keyboard } from '../../components';

const mockedCallback = jest.fn();

describe('MODULE | Keyboard', () => {
    it('component renders correctly', () => {
        const tree = render(
            <ThemeProvider>
                <Keyboard
                    focusedInput=""
                    height={300}
                    setValues={mockedCallback}
                    onSubmit={mockedCallback}
                />
            </ThemeProvider>,
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('return undefined if there is no focused input provided', async () => {
        const tree = render(
            <ThemeProvider>
                <Keyboard
                    focusedInput=""
                    height={300}
                    setValues={mockedCallback}
                    onSubmit={mockedCallback}
                />
            </ThemeProvider>,
        );
        const button = await tree.findByText('1');

        await act(async () => {
            return fireEvent.press(button);
        });
        expect(mockedCallback).not.toHaveBeenCalled();
    });
    it('sets the value on numeric button press', async () => {
        const currentState = { focusedInput: '' };
        const focusedInput = 'focusedInput';
        let nextState: any;
        const mockSetter = jest.fn().mockImplementation((callback) => {
            nextState = callback(currentState);
        });

        const tree = render(
            <ThemeProvider>
                <Keyboard
                    focusedInput={focusedInput}
                    height={300}
                    setValues={mockSetter}
                    onSubmit={() => {
                        return;
                    }}
                />
            </ThemeProvider>,
        );

        const button = await tree.findByText('1');

        await act(async () => fireEvent.press(button));

        expect(mockSetter).toHaveBeenCalled();
        expect(nextState).toEqual({ focusedInput: '1' });
    });
    it('deletes the value on delete button press', async () => {
        const currentState = { focusedInput: '' };
        const mockedTestID = 'deleteButtonTestID';
        const focusedInput = 'focusedInput';
        let nextState: any;
        const mockedSetter = jest.fn().mockImplementation((callback) => {
            nextState = callback(currentState);
        });

        const tree = render(
            <ThemeProvider>
                <Keyboard
                    deleteButtonTestID={mockedTestID}
                    focusedInput={focusedInput}
                    height={300}
                    setValues={mockedSetter}
                    onSubmit={() => {
                        return;
                    }}
                />
            </ThemeProvider>,
        );
        const button = await tree.findByTestId(mockedTestID);

        await act(async () => fireEvent.press(button));

        expect(mockedSetter).toHaveBeenCalled();
        expect(nextState).toEqual({ focusedInput: '' });
    });
    it('fires callback onSubmit press', async () => {
        const mockedSetter = jest.fn();
        const mockedOnSubmit = jest.fn();
        const mockedTestID = 'submitButtonTestID';
        const focusedInput = 'input';
        const tree = render(
            <ThemeProvider>
                <Keyboard
                    submitButtonTestID={mockedTestID}
                    focusedInput={focusedInput}
                    height={300}
                    setValues={mockedSetter}
                    onSubmit={mockedOnSubmit}
                />
            </ThemeProvider>,
        );
        const button = await tree.findByTestId(mockedTestID);

        await act(async () => fireEvent.press(button));

        expect(mockedOnSubmit).toHaveBeenCalled();
    });
});
