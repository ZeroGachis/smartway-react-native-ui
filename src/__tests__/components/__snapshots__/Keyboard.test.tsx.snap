// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MODULE | Keyboard component renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "height": 300,
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
            "height": "25%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "backgroundColor": "#ffffff",
                "flex": 1,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            style={
              {
                "color": "#2f4158",
                "fontSize": 24,
                "textAlign": "center",
              }
            }
          >
            1
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "backgroundColor": "#ffffff",
                "flex": 1,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            style={
              {
                "color": "#2f4158",
                "fontSize": 24,
                "textAlign": "center",
              }
            }
          >
            2
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "backgroundColor": "#ffffff",
                "flex": 1,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            style={
              {
                "color": "#2f4158",
                "fontSize": 24,
                "textAlign": "center",
              }
            }
          >
            3
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "height": "25%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "backgroundColor": "#ffffff",
                "flex": 1,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            style={
              {
                "color": "#2f4158",
                "fontSize": 24,
                "textAlign": "center",
              }
            }
          >
            4
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "backgroundColor": "#ffffff",
                "flex": 1,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            style={
              {
                "color": "#2f4158",
                "fontSize": 24,
                "textAlign": "center",
              }
            }
          >
            5
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "backgroundColor": "#ffffff",
                "flex": 1,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            style={
              {
                "color": "#2f4158",
                "fontSize": 24,
                "textAlign": "center",
              }
            }
          >
            6
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "height": "25%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "backgroundColor": "#ffffff",
                "flex": 1,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            style={
              {
                "color": "#2f4158",
                "fontSize": 24,
                "textAlign": "center",
              }
            }
          >
            7
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "backgroundColor": "#ffffff",
                "flex": 1,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            style={
              {
                "color": "#2f4158",
                "fontSize": 24,
                "textAlign": "center",
              }
            }
          >
            8
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "backgroundColor": "#ffffff",
                "flex": 1,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            style={
              {
                "color": "#2f4158",
                "fontSize": 24,
                "textAlign": "center",
              }
            }
          >
            9
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "height": "25%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "backgroundColor": "#ffffff",
                "flex": 1,
                "justifyContent": "center",
              },
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight="24"
              bbWidth="24"
              fill="#919eab"
              focusable={false}
              height="24"
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "flex": 0,
                    "height": 24,
                    "width": 24,
                  },
                ]
              }
              vbHeight={1024}
              vbWidth={1024}
              width="24"
            >
              <RNSVGGroup
                fill={
                  {
                    "payload": 4287733419,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  d="M432.090 645.513c7.821 7.821 16.947 11.733 27.375 11.733 10.432 0 19.558-3.913 27.379-11.733l80.179-79.202 80.179 80.179c7.821 7.821 16.785 11.733 26.889 11.733s19.063-3.913 26.889-11.733c7.168-7.821 10.756-16.947 10.756-27.379 0-10.428-3.588-19.554-10.756-27.379l-80.179-79.198 79.198-79.202c7.172-7.82 10.756-16.783 10.756-26.887s-3.584-19.066-10.756-26.889c-7.821-7.822-16.947-11.733-27.375-11.733-10.432 0-19.558 3.911-27.379 11.733l-78.221 79.201-81.156-80.178c-7.821-7.823-16.785-11.57-26.889-11.244s-19.068 4.074-26.889 11.244c-7.823 7.822-11.734 16.785-11.734 26.889s3.911 19.067 11.734 26.888l81.156 80.179-81.156 79.198c-7.171 7.825-10.756 16.951-10.756 27.379 0 10.432 3.585 19.23 10.756 26.402zM378.311 810.756c-19.556 0-37.482-4.727-53.778-14.178s-30.311-21.999-42.045-37.645l-136.889-192.623c-11.733-14.993-17.6-32.593-17.6-52.8s5.867-38.46 17.6-54.754l136.889-192.623c11.733-16.296 25.748-29.17 42.045-38.622s34.222-14.178 53.778-14.178h396.976c25.425 0 47.262 9.126 65.515 27.378 18.249 18.252 27.375 40.089 27.375 65.511v412.622c0 25.421-9.126 47.095-27.375 65.024-18.253 17.924-40.090 26.889-65.515 26.889h-396.976zM222.845 512.533l144.711 206.31h407.732v-412.622h-407.732l-144.711 206.311zM775.287 512.533v-206.311 206.311z"
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "stroke",
                      "strokeWidth",
                    ]
                  }
                  stroke={
                    {
                      "payload": 4287733419,
                      "type": 0,
                    }
                  }
                  strokeWidth="5"
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "backgroundColor": "#ffffff",
                "flex": 1,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            style={
              {
                "color": "#2f4158",
                "fontSize": 24,
                "textAlign": "center",
              }
            }
          >
            0
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "backgroundColor": "#ffffff",
                "flex": 1,
                "justifyContent": "center",
              },
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight="24"
              bbWidth="24"
              fill="#18a586"
              focusable={false}
              height="24"
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "flex": 0,
                    "height": 24,
                    "width": 24,
                  },
                ]
              }
              vbHeight={1024}
              vbWidth={1024}
              width="24"
            >
              <RNSVGGroup
                fill={
                  {
                    "payload": 4279805318,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  d="M504.802 739.021c-9.126-9.775-13.854-20.86-14.178-33.242-0.329-12.386 4.399-23.142 14.178-32.269l111.467-112.444h-484.98c-13.037 0-23.956-4.399-32.755-13.201-8.8-8.798-13.2-19.716-13.2-32.755 0-13.035 4.4-23.953 13.2-32.755 8.8-8.798 19.718-13.201 32.755-13.201h484.98l-111.467-112.443c-9.779-9.126-14.507-20.37-14.178-33.734 0.324-13.363 5.052-24.282 14.178-32.756 9.122-9.777 19.878-14.504 32.265-14.178s23.142 5.052 32.269 14.178l192.619 192.624c4.565 4.561 7.987 9.613 10.27 15.155 2.278 5.538 3.422 11.243 3.422 17.109s-1.143 11.571-3.422 17.114c-2.283 5.538-5.705 10.59-10.27 15.155l-191.642 191.642c-9.126 9.126-20.045 13.525-32.755 13.201-12.715-0.324-23.633-4.727-32.755-13.201zM900.8 772.267c-13.687 0-24.934-4.399-33.732-13.201-8.802-8.798-13.201-19.716-13.201-32.755v-424.355c0-13.037 4.399-23.955 13.201-32.755 8.798-8.8 20.045-13.2 33.732-13.2 13.039 0 23.957 4.4 32.755 13.2 8.802 8.8 13.201 19.718 13.201 32.755v424.355c0 13.039-4.399 23.957-13.201 32.755-8.798 8.802-19.716 13.201-32.755 13.201z"
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "stroke",
                      "strokeWidth",
                    ]
                  }
                  stroke={
                    {
                      "payload": 4279805318,
                      "type": 0,
                    }
                  }
                  strokeWidth="5"
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
